docs/advanced/performance.md

```diff
@@ -68,26 +52,40 @@ Thus it is possible to render a scatterplot layer with 10M items with reasonable
+
+
+## Layer Precision
+
+* 32 bit precision is the default in most layers which means that there is a limit to precision that usually becomes evident at high zoom levels (objects start to "wobble" at around mercator zoom 18). These precision can be worked around by setting the `fp64` prop to true, or by using an offset mode based coordinate system (assumes data is clusted around a local center). Flipping on 64 bit precision affects performance as described above and in the separate chapter about the 64-bit GPU solution in deck.gl.
 
 
 ## Layer Picking Performance
 
-Recommendation: Enabling picking has a small performance penalty so make sure
-the `pickable` property is `false` in layers that do not need picking (this
-is the default value).
+deck.gl performs picking by drawing the layer into an off screen picking buffer. This essentially means that every layer that supports picking will be drawn off screen when panning and hovering. The picking is performed using the same GPU code that does the visual rendering, so the performance should be easy to predict.
 
-deck.gl performs picking by drawing the layer into an off screen
-picking buffer. This essentially means that every layer that supports picking
-will be drawn off screen when panning and hovering.
+Picking limitations:
+* The picking system can only distinguish between 16M items per layer.
+* The picking system can only handle 256 layers with the pickable flag set to true.
 
 
-## Layer Precision
+## Number of Layers
+
+The layer count of an advanced deck.gl application tends to gradually increase, especially when using composite layers. We have built and optimized a highly complex application using close to 100 deck.gl layers (this includes hierarchies of sublayers rendered by custom composite layers rendering other composite layers) without seeing any performance issues related to the number of layers. If you really need to, it is probably possible to go a little higher (a few hundred layers). Just keep in mind that deck.gl was not designed to be used with thousands of layers.
+
+
+## Profiling Ideas
+
+Some profiling techniques:
+* Using the `seer` chrome extension you can also get GPU timings for your layers.
+* The `layer-browser` example (in the `examples` folder has a couple of performance tests that you can use to get FPS readings on your hardware for Scatterplot layers with 1M and 10M points.
+
+## Common Issues
+
+A couple of particular things to watch out for that tend to have a big impact on performance:
+* Be careful when enabling Retina/High DPI rendering. It generetes 4x the number of pixels (fragments) and can have a big performance impact that depends on which computer or monitor is being used.
```

Comment:
> Unrelated: is it necessary to draw the picking buffer in high DPI? Mouse coordinates are always in screen pixels.

So what we are discussing now is to make it easy for the app to select whether to draw in High-DPI or not (since it has a significant perf impact), and maybe even add a checkbox to the examples enabling the user to try with and without.

I would expect that to be a new prop on the `DeckGL` (`useHighDPI`, `useDevicePixelRation`, `useRetina` or similar) and it would affect both normal render and picking.

Once that is done, a second reasonable question is whether we could use different values for picking and main rendering. For pixel perfect picking an app might want them to be the same, but many apps want to use a picking radius for better fat finger sensitivity and then it might not matter anyway. 

I will add these comments to the open ticket #843.